tests/models/test_gpu.py

```diff
@@ -93,6 +100,32 @@ def test_multi_gpu_model_dp(tmpdir):
     memory.get_memory_profile('min_max')
 
 
+@pytest.mark.parametrize('cli_args', [
+    pytest.param('--max_epochs 1 --gpus 2 --distributed_backend ddp'),
+])
+@pytest.mark.parametrize('variation', train_test_variations.get_variations())
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason="test requires multi-GPU machine")
+def test_multi_gpu_model_ddp(tmpdir, cli_args, variation):
```

Comment:
Can you add a whole lot of comments to this test. There's a lot of non obvious things. E.g.

Why do you need to create a subprocess to test? I'm assuming it's so the ddp processes close down correctly?
Why do you need to set the PYTHONPATH?

Reply:
yeah, this pytest doesn't work well with ddp. So far this was my best idea to launch a subprocess

> Why do you need to set the PYTHONPATH?

took me very long to figure out. basically CI does not do pip install pytorch lightning, instead it launches it with 
python -m pytorch_lightning etc.